src/state_machine/signer/mod.rs

```diff
@@ -344,6 +406,10 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
     }
 
     /// Process the slice of packets
+    ///
+    /// # Panics
+    /// This function will panic if message signing fails, which can occur if the
+    /// `network_private_key` is invalid or not properly initialized.
```

Comment:
We can also make this function no panic and just return an error if it fails to sign.

Reply:
One longstanding problem with this function is that if there is a failure part way through, it does not notify the caller which messages were still in the queue at the time of the error.

`sBTC` does not use this function, so maybe we should just remove it?

Reply:
I vote to cfg test